lib/directory-monitor-poll.c

```diff
@@ -86,14 +104,28 @@ static void
 _rescan_directory(DirectoryMonitorPoll *self)
 {
   GError *error = NULL;
+  gchar *fullpath = NULL;
+  GStatBuf stat_result;
   GDir *directory = g_dir_open(self->super.real_path, 0, &error);
   collection_comparator_start(self->comparator);
+
   if (directory)
     {
       const gchar *filename;
       while((filename = g_dir_read_name(directory)))
         {
-          collection_comparator_add_value(self->comparator, filename);
+          fullpath = build_filename(self->super.real_path, filename);
```

Comment:
I have not checked the wildcard file source implementation in detail, but it seems that fullpath is needed for ``g_stat``. As far as I remember - it didn't work otherwise. 
[https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-stat](url)

Reply:
Can you highlight the sentence that writes we need full path? I must be overlooking something, but I cannot find it.

I tried with a small c code, and g_stat works for me for relative paths:
```
$ touch relative_path.txt; cat | gcc -xc - `pkg-config glib-2.0 --cflags` `pkg-config glib-2.0 --libs` && ./a.out
#include <glib/gstdio.h>
#include <assert.h>

int main() {
  GStatBuf result;
  assert(0 == g_stat("./relative_path.txt", &result));
  printf("size: %lu\n", result.st_mtime);
};
size: 1546934223
$ 
```

Reply:
you're right I need to take a deeper look on this.

Reply:
I remember now. I think the problem was that the existing code only had 'filename' in each iteration, but ``g_stat`` needs a path (or a relative path) - which would be different per platform (windows/*nix) and thus, current working directory path must actually be built.

Thus, the code modification below gave errors like: `` [2019-01-08T05:54:31.725568] Cannot stat file/directory; filename='syslog-ng.conf' `` 
```
  if (directory)
    {
      const gchar *filename;
      while((filename = g_dir_read_name(directory)))
        {
          if (g_stat(filename, &stat_result))
            {
              msg_debug("Cannot stat file/directory",
                        evt_tag_str("filename", filename));
            }
          else
            {
              collection_comparator_add_value(self->comparator, filename, stat_result.st_mtime);
            }
        }
      g_dir_close(directory);
      collection_comparator_stop(self->comparator);

```